### PR TITLE
User removes items

### LINKED ISSUE #6: Unauthenticated User can remove items from their cart
As an Unauthenticated User
When I visit /cart
And I click “Remove from Cart” on an Item
And I see a flash alert that “The item has been removed from your cart”
And I see the item disappear from my Cart
Then I should see the cart size decrease by one item

### LINKED ISSUE #9: Unauthenticated User can increase the quantity of a item in their cart
As an Unauthenticated User
When I visit /cart
And I increase the quantity of an item in my cart
Then the size of my cart should increase by one

### PR DESCRIPTION
**Pull request by teresafinn**
Closes #9 and Closes #6

### CODE REVIEW

File: app/controllers/cart_items_controller.rb [hunk 1]
```diff
@@ -4,11 +4,21 @@ def create
     @cart.add_item(item)
 
     session[:cart] = @cart.data
-    flash[:notice] = "You have #{pluralize(@cart.item_quantity(item), item.name)} in your cart."
-    redirect_to menu_path
+    flash[:notice] =
+    "You have #{pluralize(@cart.item_quantity(item), item.name)} in your cart."
+    redirect_to :back
```
Comment (indiesquidge) [inline app/controllers/cart_items_controller.rb]: What does this redirect to?

File: app/views/cart_items/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 <h1>Your Cart</h1>
 
     <% @cart.display_items.each do |item,quantity| %>
-      <%= item.name %>
+      <p><%= item.name %><br>
       Quantity: <%= quantity %><br/>
+      <%= link_to "Remove from Cart", cart_item_path(id: item.id), method: :delete %><br>
+        <%= button_to "Add Another", cart_items_path(item: item) %></br>
+      </p>
```
Comment (indiesquidge) [inline app/views/cart_items/index.html.erb]: Fix the indentation alignments here. Should be
```erb
<p>
  <%= item.name %><br>
  Quantity: <%= quantity %><br/>
  <%= link_to "Remove from Cart", cart_item_path(id: item.id), method: :delete %><br>
  <%= button_to "Add Another", cart_items_path(item: item) %></br>
</p>
```

File: config/routes.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
     resources :categories, param: :category_name
   end
 
-  resources :cart_items, only: [:create]
+  resources :cart_items, only: [:create, :destroy]
```
Comment (indiesquidge) [inline config/routes.rb]: Nice. I like that the limitation is placed as far down as the route.

File: spec/users/user_manages_cart_spec.rb [hunk 4]
```diff
@@ -35,22 +27,23 @@
...
+  it "allows users to increase the quantity of an item in their cart" do
+    create(:item)
+    visit "/menu"
+    click_link_or_button "Add to Cart"
+    visit "/cart"
+    click_link_or_button "Add Another"
+    expect(page).to have_content("Quantity: 2")
+    expect(page).to have_content("Total Items: 2")
+  end
 end
```
Comment (indiesquidge) [inline spec/users/user_manages_cart_spec.rb]: As Josh pointed out, all the tests in this file start with the exact same lines of code. We need to find a way to abstract that away.

Comment (teresafinn) [inline spec/users/user_manages_cart_spec.rb]: Hoping for help on this later today, havent found a way to stub correctly yet!

File: app/views/cart_items/index.html.erb (hunk 2 above)
Comment (teresafinn) [inline app/views/cart_items/index.html.erb]: Changed!

**Pull request merged**